x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1786,11 +1788,11 @@ private void searchActiveTokens(
      */
     private void sourceIndicesWithTokensAndRun(ActionListener<List<String>> listener) {
         final List<String> indicesWithTokens = new ArrayList<>(2);
-        final SecurityIndexManager frozenTokensIndex = securityTokensIndex.freeze();
+        final SecurityIndexManager frozenTokensIndex = securityTokensIndex.defensiveCopy();
         if (frozenTokensIndex.indexExists()) {
             // an existing tokens index always contains tokens (if available and version allows)
-            if (false == frozenTokensIndex.isAvailable()) {
-                listener.onFailure(frozenTokensIndex.getUnavailableReason());
+            if (false == frozenTokensIndex.isAvailable(PRIMARY_SHARDS)) {
+                listener.onFailure(frozenTokensIndex.getUnavailableReason(PRIMARY_SHARDS));
                 return;
             }
```

Comment:
I think the availability checks in the `sourceIndicesWithTokensAndRun` should look for the "search" variant, since that's how the indices are used for in the `searchActiveTokens` method. Also maybe include the "search" name somewhere in the method name.